cmd/skaffold/app/cmd/dev.go

```diff
@@ -109,9 +110,13 @@ func dev(out io.Writer, ui bool) error {
 			}
 			if err != nil {
 				if errors.Cause(err) != runner.ErrorConfigurationChanged {
+					plugin.CleanupClients()
```

Comment:
just a question -- why did you opt to have these twice and remove the defer function?

Reply:
because you're not suppose to call defer in a for loop this way - it might actually end up causing problems across dev cycles. one nice explanation is here: https://blog.learngoprogramming.com/gotchas-of-defer-in-go-1-8d070894cb01 - I could have switched over to an anonymous function as well...